engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java

```diff
@@ -2794,6 +2814,39 @@ public static Instant lowerBin(@Nullable final Instant instant, long intervalNan
         return epochNanosToInstant(Numeric.lowerBin(epochNanos(instant) - offset, intervalNanos) + offset);
     }
 
+    /**
+     * Returns an {@link Instant} value, which is at the starting (lower) end of a time range defined by the interval
+     * nanoseconds. For example, a 5*MINUTE intervalNanos value would return the instant value for the start of the
+     * five-minute window that contains the input instant.
+     *
+     * @param instant instant for which to evaluate the start of the containing window
+     * @param intervalNanos size of the window in nanoseconds, either as an integer value or as a time interval string,
+     *        e.g. "PT00:00:00.001" or "PT1s"
+     * @param offset The window start offset in nanoseconds, either as an integer value or as a time interval string,
+     *        e.g. "PT00:00:00.001" or "PT1s". For example, a value of MINUTE would offset all windows by one minute.
+     * @return {@code null} if either input is {@code null}; otherwise, an {@link Instant} representing the start of the
+     *         window
+     */
+    @ScriptApi
+    @Nullable
+    public static Instant lowerBin(@Nullable final Instant instant, Object intervalNanos, Object offset) {
+        long interval = getDurationNanos(intervalNanos);
+        long off = getDurationNanos(offset);
+
+        return lowerBin(instant, interval, off);
+    }
```

Comment:
This new overload is to support `Union` type in the Python API, it will not conflict with the original overload that takes primitive type parameters. The other approach would be to create multiple overload methods for all the possible combinations of parameter types.

Reply:
An overload per type is the Java way. This current way is not usable from Java. If we want it, at least hide it in some other class that we never tell Java programmers about.

Reply:
Maybe the answer looks something like applying a conversion to the input and calling the existing overload? That said, if the input is from a column, for example, that doesn't really work.